lib/ecto/adapter/migration.ex

```diff
@@ -50,4 +50,7 @@ defmodule Ecto.Adapter.Migration  do
     * `:log` - When false, does not log begin/commit/rollback queries
   """
   @callback execute_ddl(repo :: Ecto.Repo.t, command, options :: Keyword.t) :: :ok | no_return
+
+  @callback lock_for_migrations(repo :: Ecto.Repo.t, table :: String.t, options :: Keyword.t, function :: fun) ::
+    {:ok, any} | {:error, any}
```

Comment:
Here you say it can return `{:error, _}` but when you call it in migration you don't check for the error tuple. Should we always say it returns `{:ok, any}` then?

Reply:
We could also say it simply returns `any`. i.e. we don't expect the operation to fail. Or it returns the value or it raises.